Updates to Proof System Docs (#1361)
risc0/risc0: website/docs/proof-system/proof-system.md

# Proof System Overview

<center> <img src="assets/receipt.png" width="600"/> </center>
<br/>
When the RISC Zero [zkVM] executes, it produces a [Receipt] that serves as a proof of validity of a given [Session].

To confirm that a [Receipt] was honestly generated, use [Receipt::verify()].

The [Receipt] includes a [journal], which contains the public outputs of the [Session].
The contents of the [journal] are specified by calling [env::commit()] and [env::commit_slice()] in the [guest] code.

In addition to the journal, a receipt makes a number of claims about the program execution.
These claims are summarized in the [ReceiptClaim].

### Types of Receipts
A [Receipt] can take four main forms.

> 1. A **[flat receipt]** is a vector of [ZK-STARK]s, one for each [segment]. Segments & segment proofs are constructed by the [RISC-V Circuit].

Reviewer: We renamed this to "Composite" receipt

Author: i.e. for 0.20?

Reviewer: Checking with @nategraf but I thought we did that in `0.19`

Author: I wrote this based on what's written [here](https://docs.rs/risc0-zkvm/0.19.1/risc0_zkvm/enum.InnerReceipt.html)

Reviewer: OK, so the `docs/` directory is considered (unversioned), so I'm not sure how we should deal with name changes like this.

Author: I'm not strongly opinionated, but I guess it seems nice to me to maintain versioned history for these proof system docs.

The [Receipt] includes a [journal], which contains the public outputs of the [Session].
The contents of the [journal] are specified by calling [env::commit()] and [env::commit_slice()] in the [guest] code.

In addition to the journal, a receipt makes a number of claims about the program execution.
These claims are summarized in the [ReceiptClaim].

### Types of Receipts

A [Receipt] can take four main forms.

1. A **[flat receipt]** is a vector of [ZK-STARK]s, one for each [segment]. Segments & segment proofs are constructed by the [RISC-V Circuit].
2. A **[succinct receipt]** is a single [ZK-STARK] proving an entire [Session]. A succinct receipt is formed by aggregating the proofs from a flat receipt. This is accomplished using the [Recursion Circuit]. Users can also aggregate multiple succinct receipts into a single succinct receipt using [proof composition].
3. A **[Groth16 receipt]** is a single [Groth16] proof for an entire [Session]. A Groth16Receipt is formed by verifying a succinct receipt using RISC Zero's [Groth16 circuit].
4. A **[fake receipt]** doesn't contain any proof at all. This feature is offered to enable rapid prototyping. See [Dev Mode] for more information.

> <center> <img src="assets/proof-system-layers.png"/> </center>

Author: ```suggestion
![The layers of the RISC Zero proving system](assets/proof-system-layers.png)
```

# Proof System Overview

> <center> <img src="assets/receipt.png" width="600"/> </center>
> <br/>

Reviewer 2: Going by how our STARK By Hand docs work, I think we can do this like
```suggestion
![A diagram of a receipt as the output of a zkVM program](assets/receipt.png)
```

Author: Thanks for the suggestion -- the place I'm feeling stuck is when I try to specify a non-default size display...

Reviewer 2: Resizing sounds possible but cumbersome: https://stackoverflow.com/questions/58265585/change-image-size-in-docusaurus

Author: Hmm I took a stab at this and it wasn't immediately obvious. I'm going to go ahead and accept the suggestion you proposed and if the sizing looks bad we can fix it on a future PR
